RevenueCatUI/Views/PurchaseButton.swift

```diff
@@ -0,0 +1,144 @@
+//
+//  PurchaseButton.swift
+//  
+//
+//  Created by Nacho Soto on 7/18/23.
+//
+
+import RevenueCat
+import SwiftUI
+
+@available(iOS 16.0, macOS 13.0, tvOS 16.0, *)
+struct PurchaseButton: View {
+
+    let package: Package
+    let purchaseHandler: PurchaseHandler
+    let colors: PaywallData.Configuration.Colors
+    let localization: ProcessedLocalizedConfiguration
+    let introEligibility: IntroEligibilityStatus?
+    let mode: PaywallViewMode
+
+    @Environment(\.dismiss)
+    private var dismiss
+
+    var body: some View {
+        self.button
+    }
+
+    private var button: some View {
+        AsyncButton {
+            let cancelled = try await self.purchaseHandler.purchase(package: self.package).userCancelled
+
+            if !cancelled, case .fullScreen = self.mode {
+                self.dismiss()
+            }
+        } label: {
+            IntroEligibilityStateView(
+                textWithNoIntroOffer: self.localization.callToAction,
+                textWithIntroOffer: self.localization.callToActionWithIntroOffer,
+                introEligibility: self.introEligibility
+            )
+            .foregroundColor(self.colors.callToActionForegroundColor)
+            .tint(self.colors.callToActionForegroundColor)
+            .frame(
+                maxWidth: self.mode.fullWidthButton
+                ? .infinity
+                : nil
+            )
+        }
+        .font(self.mode.buttonFont.weight(.semibold))
+        .tint(self.colors.callToActionBackgroundColor.gradient)
+        .buttonBorderShape(self.mode.buttonBorderShape)
+        .controlSize(self.mode.buttonSize)
+        .buttonStyle(.borderedProminent)
+        .frame(maxWidth: .infinity)
+    }
+
+}
+
+@available(iOS 16.0, macOS 13.0, tvOS 16.0, *)
+private extension PaywallViewMode {
+
+    var buttonFont: Font {
+        switch self {
+        case .fullScreen, .card: return .title2
+        case .banner: return .footnote
+        }
+    }
```

Comment:
how do you envision this scaling to different templates?

Reply:
Probably would need to inject this font to `PurchaseButton` instead.